shell/platform/linux/fl_accessible_node_test.cc

```diff
@@ -0,0 +1,79 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+// Included first as it collides with the X11 headers.
+#include "gtest/gtest.h"
+
+#include "flutter/shell/platform/linux/fl_accessible_node.h"
+#include "flutter/shell/platform/linux/testing/fl_test.h"
+
+// Checks can build a tree of nodes.
+TEST(FlAccessibleNodeTest, BuildTree) {
+  g_autoptr(FlEngine) engine = make_mock_engine();
+
+  g_autoptr(FlAccessibleNode) root = fl_accessible_node_new(engine, 0);
+  g_autoptr(FlAccessibleNode) child1 = fl_accessible_node_new(engine, 1);
+  fl_accessible_node_set_parent(child1, ATK_OBJECT(root));
+  g_autoptr(FlAccessibleNode) child2 = fl_accessible_node_new(engine, 1);
+  fl_accessible_node_set_parent(child2, ATK_OBJECT(root));
+  g_autoptr(GPtrArray) children =
+      g_ptr_array_new_with_free_func(g_object_unref);
+  g_ptr_array_add(children, g_object_ref(child1));
+  g_ptr_array_add(children, g_object_ref(child2));
+  fl_accessible_node_set_children(root, children);
```

Comment:
Are there any test expectations we can add here?

Reply:
Added some ATK calls to confirm the parents and children are set as we expect.